### DISCUSSION TITLE
Wierd-but-valid

### DISCUSSION THREAD

**Discussion by cherylcarpenter** [tailwindlabs/tailwindcss#5781] [Help]
why does JIT plus purge produce this??:

.w-\[this-is\\\\\]w-\\\\\[weird-but-valid\] {
	width: this-is\\]w-\\[weird-but-valid;
}

which causes project build to fail.

**Comment by GrahamSH-LLK** [comment]
I assume you have a line somewhere that has that code somewhere

**Comment by sachinraja** [comment]
My guess is that your content setting includes `node_modules` (it is not specific enough) and it matched the class name here:
https://github.com/tailwindlabs/tailwindcss/blob/6ce8864f47061035cd79b2d563eea72467dcb3b7/src/lib/setupContextUtils.js#L118-L120

More info on this in [the docs](https://tailwindcss.com/docs/content-configuration#pattern-recommendations). It would help if you shared your tailwind config file.

**Reply by adamwathan** [reply]
Yep this is the reason for sure 👍🏻

**Reply by yorch** [reply]
thanks! yes, exactly what was happening in my case

**Reply by shawninder** [reply]
I'm not sure I understand. If it matched that class name and included it in the CSS, why would that break the other CSS declarations below? Is TailwindCSS injecting an invalid class name? Why would it match an invalid class name?

I do get the part about not including node_modules, of course, but I'm afraid to run into the same problem with a rogue comment in my own code unless I fully understand why this is happening…

**Reply by sachinraja** [reply]
Yes, it is producing invalid css, this class name has an [arbitrary value](https://tailwindcss.com/docs/adding-custom-styles#using-arbitrary-values), so it can really be anything.

**Comment by RyanGaudion** [comment]
Same issue here, tailwind 3 produces this line:

```
.w-\[this-is\\\\\]w-\\\\\[weird-but-valid\] {
  width: this-is\\]w-\\[weird-but-valid;
}

```
however, this line breaks the entire css file and causes my page to render without css. I know I should ignore the node_module folder but is there a way to stop this from breaking the output css?

**Reply by anesdiaf** [reply]
@RyanGaudion I know my response maybe late but, the solution to these is to specify your content, which is your templates location & file types in the tailwind.config.js file.

**Comment by Triloworld** [comment]
Problem:
PostCSS don't discard invalid css styles

Temporary solution for this case:
File: tailwind.config.js
Edit part of code with content:
```js
module.exports = {
  content: [
    './**/*.{html,php,vue,jsx,tpl}',
    './assets/**/*.{html,php,vue,js,jsx,tpl}'
  ],
  safelist: [
    {
      pattern: /./ 
    },
  ],
}
```

JS file cause error in node_modules:
https://github.com/tailwindlabs/tailwindcss/blob/6ce8864f47061035cd79b2d563eea72467dcb3b7/src/lib/setupContextUtils.js#L118-L120
```css
.w-\[this-is\\\\\]w-\\\\\[weird-but-valid\] {
  width: this-is\\]w-\\[weird-but-valid;
}
```

Proper solution propose:
Option 1: Remove part of this code from source and mark it as invalid
Option 2: Fix this code to be proper for browser interpreter
Option 3: Upstream fix to PostCSS to discard any not valid css and document that validation of css in required after tailwind process
Option 4: Add to tailwind fix for proper manage that case (bad option - this can be added as PostCSS plugin and add to documentation about warning that code isn't guaranteed to be valid)
Option 5: This is same as node_modules in whole added in tailwind to scan. Maybe option that need to be toggled on to scan files in node_modules? I prefer to add warning (and option to disable warning), this ensure great DX, close this topic and give option to show warning if node_module directory in use that will break your code if not used proper.

My view:
Tailwind add it even if not correct to css file and no warning. There can be small fix but introduce whole lot new one as this isn't validation processor but code generation. *I fix it by narrow scope of content* but in first touch with error is not that easily to spot ;/ more warning can be added to spot if tailwind used not in way as documented and proper.
Code from tailwind not guaranteed  to be valid, looking for tester for PostCSS who can check cases like that (every one use external css styles and this help ensure that code maintain only valid code and warm if not)
